expression/aggregation/descriptor.go

```diff
@@ -39,6 +40,8 @@ type AggFuncDesc struct {
 	Mode AggFunctionMode
 	// HasDistinct represents whether the aggregation function contains distinct attribute.
 	HasDistinct bool
+	// SessionCtx tracks the session context of the AggFunc.
+	SessionCtx sessionctx.Context
```

Comment:
I think we may be better not define SessionCtx as an attribute of AggFuncDesc.
We pass SessionCtx as an input parameter in the functions which need it.